java/openpkw-services/src/main/java/org/openpkw/services/CsvParser.java

```diff
@@ -0,0 +1,37 @@
+package org.openpkw.services;
+
+import org.springframework.batch.item.file.LineMapper;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by wojciech.szostak on 15.08.2015.
+ */
+public class CsvParser<T>
+{
+    private LineMapper lineMapper;
+
+    public CsvParser(LineMapper<T> lineMapper) {
+        this.lineMapper = lineMapper;
+    }
+
+    public List<T> parse(File file) throws Exception
+    {
+        List<T> result = new ArrayList<>();
+
+        BufferedReader br = new BufferedReader(new FileReader(file));
+
+        int lineNumber = 0;
+        for(String line; (line = br.readLine()) != null; ) {
```

Comment:
To jest w pętli. Wyciągnięcie tego do jakiejś metody chyba nie poprawi czytelności. Sprawdzam `nulla` bo `readLine` to zwraca jeżeli nie ma już nic więcej do czytania. Teoretycznie pusty string jest poprawną wartością